lib/codec.js

```diff
@@ -88,6 +93,29 @@
     return _.compose(updateEncOffset(bytebuf), checkedWriteSignedInt(bytebuf), zigZag)(num);
   }
 
+  function doEncodeProtobuf(bytebuf, message, typeId) {
+    return doEncodeBytesWithLength(bytebuf,encodeProtobuf(message,typeId));
+  }
+
+  function encodeProtobufInstance(root){
+    return function(obj){
+      return root.encode(obj).finish();
+    }
+  }
+
+  function encodeProtobuf(message,typeId){
+    var encodedMessage = encodeProtobufInstance(message.$type)(message);
+    var messageTypeId = typeId.apply(null,[message.$type.fullName]);
```

Comment:
Is this preferable to `typeId(message.$type.fullName)` ?

Reply:
I don't have such preference. Just wanted to do this in fp so written in that way.
Should I change that?

Reply:
Why do you consider `.apply`more fp oriented than calling `typeId` directly?
I would say that a direct call to `typeId` is more readable.

Reply:
Ok sir. I will change that.